### ISSUE TITLE
HAVCAP mission

### ISSUE THREAD

**Issue by DanAlbert** [dcs-liberation/dcs_liberation#843]
Needed now that we have (currently optional) non-invincible AEW&C missions.

Unsure if we're going to be planning 4 hour missions for the aircraft or if we need an overhaul to allow multiple flights to be scheduled over the course of the asset's mission.

**Comment by SnappyComebacks** [comment]
Just messed around in the editor with this idea.

If you set the HAVCAP flight to have the following actions (in order):

1. `CAP`
2.  Restrict Afterburner on (optional, if you don't want AB used to rejoin flight after an engagement.  Probably ideal for an orbiting escort, but not A -> B escort)
3. `Follow`

They will follow until encroached on, attack the encroaching hostile, then return back to the follow task.  Bonus feature is if the HAVCAP is on a compatible tanker, the flight will refuel when it needs to.  The bad part is with the aforementioned settings, the HAVCAP flight will use full AB when returning from `CAP` to `Follow`.

This probably can be refined further, however it's likely that multiple missions will need to be planned if the mission is expected to have any engagements.

**Comment by DanAlbert** [comment]
So the follow fast works after the tanker or AWACS starts its main task? I'd tried using follow to implement escorts but the moment the strike flight begins its attack the follow flight just wandered uselessly (same bug as the escort task mentioned on the front page). I'd buy that refuel and AWACS don't have that problem, just want to be sure.

You could maybe disable AB after they're on station. Probably best to just not worry about it at first though. It may not be a big deal.

**Comment by SnappyComebacks** [comment]
My test mission had all flights start in air already.  Blufor flying from over the Black Sea to land at Tbilisi.

- KC135
  - Tanker task enabled at mission start (no racetrack)
- 2x F15Cs
  - Default tasks (CAP is very first item)
  - Follow task on KC135 at end of default task list (To follow the KC135)
- Mig-29 with waypoints forcing intercept with KC135
  - CAP task

So I don't know if it would behave different starting from cold and dark. 

I also don't know if disabling AB would also disable the flights use during combat, which would be bad.

**Comment by DanAlbert** [comment]
> So I don't know if it would behave different starting from cold and dark.

I don't think so. The escort follow problem happened with cold starts and in-air. I think as long as your test was doing the refueling and orbit (racetrack) tasks, that's all that matters. Ditto for AWACS and orbit (circle). Sounds like that's the case, which is good news!
